include/oneapi/dpl/internal/gcd_impl.h

```diff
@@ -27,6 +27,12 @@ namespace dpl
 
 namespace internal
 {
+// We need have own implementation instead of std::gcd and std::lcm because
```

Comment:
1) I would place the comment directly before gcd and lcm implementations.
2) I think we should write about an error.. A short explanation why and "reference" (not link) to the C++ limitations in SYCL spec.
f.e. "SYCL doesn't  support recursive call in a kernel. For more details about C++ limitations see the SYCL specification."

Reply:
Fixed

Reply:
I think Mikhail meant to keep a single comment, but put it right before the function implementation starts. At least, I see no reason to duplicate the exact same wording twice. I would keep only the first instance, and moreover put a space line after the comment to visually separate it from the following function.

As for the wording, I would slightly change it, e.g. "Some C++ standard libraries implement std::gcd and std::lcm as recursive functions, which prevents their use in SYCL kernels (see the SYCL specification for more details). Therefore oneDPL provides its own implementation."

Reply:
Fixed